### PR TITLE
Bootstrap Modal for Model Upload Instructions

### PR DESCRIPTION
**Pull request by Navnoorsinghmahal**
## 📜 Description

This pull request implements a **Bootstrap Modal** feature for the model upload instructions in the Dashboard. The modal provides clear guidelines for users to upload models in the correct format and size, enhancing user experience and ensuring smoother interactions.

---

## ✅ Changes Made

- Added a **Bootstrap Modal** to the Dashboard for displaying upload instructions.
- Included details about:
  - **File Format**: `.pkl` files only.
  - **File Size**: Less than 1 MB.
- Integrated buttons in the modal for:
  - **Upload Model**: Triggers the file input.
  - **Close**: Closes the modal.
- Styled the modal using inline CSS for customization.
- Adjusted modal size to be compact for a better fit within the UI.

---

## ✅ Requirements Addressed

- Provide users with clear instructions on acceptable file formats and size limits.
- Ensure the modal is reusable and adheres to Bootstrap conventions.
- Maintain a clean and user-friendly interface.

---

## 📋 Acceptance Criteria

- [x] A Bootstrap Modal is present and opens when triggered.
- [x] The modal displays clear upload instructions.
- [x] Buttons in the modal function as expected:
  - "Upload Model" triggers file input and closes the modal.
  - "Close" simply closes the modal.
- [x] The modal design is compact and visually aligns with the rest of the Dashboard.

### CODE REVIEW

File: frontend/src/pages/DashboardPage/ControlButtons/ControlButtons.jsx [hunk 1]
```diff
@@ -175,6 +181,72 @@ const ControlButtons = ({ onDownload }) => {
...
+                          borderRadius: "0.2rem"  // Optional: round corners slightly
+                        }}
+                        onClick={closeModal}
+                    >
+                      &times;
+                    </button>
+                  </div>
+                  <div className="modal-body">
+                    <p><strong>File format:</strong> The file must be in <code>.pkl</code> format.</p>
+                    <p><strong>File size:</strong> The file size must be less than 1 MB.</p>
```
Comment (Jai0212) [inline frontend/src/pages/DashboardPage/ControlButtons/ControlButtons.jsx:209]: Are we checking in backend if file is more than 1 MB?

File: frontend/src/pages/DashboardPage/ControlButtons/ControlButtons.jsx [hunk 2]
```diff
@@ -57,7 +58,7 @@ const ControlButtons = ({ onDownload }) => {
   }, []);
 
   const handleImportModels = () => {
-    fileInputRef1.current.click();
+    setShowModal(true);  // Open the modal when button is clicked
```
Comment (Jai0212) [inline frontend/src/pages/DashboardPage/ControlButtons/ControlButtons.jsx:61]: Can this be implemented for upload data also?

**Review by Jai0212 — COMMENTED** [review]
Reviewed, Everything looks good

File: frontend/src/pages/DashboardPage/ControlButtons/ControlButtons.jsx (hunk 2 above)
Comment (Navnoorsinghmahal) [reply frontend/src/pages/DashboardPage/ControlButtons/ControlButtons.jsx:61]: Yes we can implement for upload data but can you specify the conditions we want to put on the bootstrap modal.

File: frontend/src/pages/DashboardPage/ControlButtons/ControlButtons.jsx (hunk 1 above)
Comment (Navnoorsinghmahal) [reply frontend/src/pages/DashboardPage/ControlButtons/ControlButtons.jsx:209]: We haven't added any such conditions yet.

File: frontend/src/pages/DashboardPage/ControlButtons/ControlButtons.jsx (hunk 2 above)
Comment (Jai0212) [reply frontend/src/pages/DashboardPage/ControlButtons/ControlButtons.jsx:61]: We will look into that as the upload dataset is quite straightforward (csv)

Comment (Navnoorsinghmahal) [reply frontend/src/pages/DashboardPage/ControlButtons/ControlButtons.jsx:61]: Will implement this also in my next pr

Comment (Jai0212) [reply frontend/src/pages/DashboardPage/ControlButtons/ControlButtons.jsx:61]: Ok

**Pull request merged**